### ISSUE TITLE
403 forbidden for shoreurl?

### ISSUE THREAD

**Issue by gmkou** [gmkou/FikaNote#3]
When i try to add "http://t.co/sJerAJfnH0" to agenda, it shows an error.

HTTPError at /add/
HTTP Error 403: Forbidden

**Comment by gmkou** [comment]
Environment:

Request Method: POST
Request URL: http://localhost:5000/add/

Django Version: 1.6.5
Python Version: 2.7.8
Installed Applications:
('django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'hello')
Installed Middleware:
('django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware')

Traceback:
File "/Users/kou/Programs/WHH/FikaNote/venv/lib/python2.7/site-packages/django/core/handlers/base.py" in get_response
1.                     response = wrapped_callback(request, _callback_args, *_callback_kwargs)
   File "/Users/kou/Programs/WHH/FikaNote/hello/views.py" in add
2.             soup = BeautifulSoup(urllib2.urlopen(url))
   File "/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py" in urlopen
3.     return _opener.open(url, data, timeout)
   File "/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py" in open
4.             response = meth(req, response)
   File "/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py" in http_response
5.                 'http', request, response, code, msg, hdrs)
   File "/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py" in error
6.         result = self._call_chain(*args)
   File "/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py" in _call_chain
7.             result = func(*args)
   File "/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py" in http_error_302
8.         return self.parent.open(new, timeout=req.timeout)
   File "/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py" in open
9.             response = meth(req, response)
   File "/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py" in http_response
10.                 'http', request, response, code, msg, hdrs)
    File "/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py" in error
11.             return self._call_chain(*args)
    File "/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py" in _call_chain
12.             result = func(*args)
    File "/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py" in http_error_default
13.         raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)

Exception Type: HTTPError at /add/
Exception Value: HTTP Error 403: Forbidden

**Comment by katoken-0215** [comment]
This is the code to reproduce problem.

```
$ python
Python 2.7.9 (default, Feb 15 2015, 00:40:58) 
[GCC 4.8.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import urllib2
>>> urllib2.urlopen("http://t.co/sJerAJfnH0")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/katoken/.pyenv/versions/2.7.9/lib/python2.7/urllib2.py", line 154, in urlopen
    return opener.open(url, data, timeout)
  File "/home/katoken/.pyenv/versions/2.7.9/lib/python2.7/urllib2.py", line 437, in open
    response = meth(req, response)
  File "/home/katoken/.pyenv/versions/2.7.9/lib/python2.7/urllib2.py", line 550, in http_response
    'http', request, response, code, msg, hdrs)
  File "/home/katoken/.pyenv/versions/2.7.9/lib/python2.7/urllib2.py", line 469, in error
    result = self._call_chain(*args)
  File "/home/katoken/.pyenv/versions/2.7.9/lib/python2.7/urllib2.py", line 409, in _call_chain
    result = func(*args)
  File "/home/katoken/.pyenv/versions/2.7.9/lib/python2.7/urllib2.py", line 656, in http_error_302
    return self.parent.open(new, timeout=req.timeout)
  File "/home/katoken/.pyenv/versions/2.7.9/lib/python2.7/urllib2.py", line 437, in open
    response = meth(req, response)
  File "/home/katoken/.pyenv/versions/2.7.9/lib/python2.7/urllib2.py", line 550, in http_response
    'http', request, response, code, msg, hdrs)
  File "/home/katoken/.pyenv/versions/2.7.9/lib/python2.7/urllib2.py", line 475, in error
    return self._call_chain(*args)
  File "/home/katoken/.pyenv/versions/2.7.9/lib/python2.7/urllib2.py", line 409, in _call_chain
    result = func(*args)
  File "/home/katoken/.pyenv/versions/2.7.9/lib/python2.7/urllib2.py", line 558, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
urllib2.HTTPError: HTTP Error 403: Forbidden
```

**Comment by katoken-0215** [comment]
t.co seems checking User-Agent.
And, its response is <meta> redirection code.
Should we introduce special handler for t.co ?

```
>>> import urllib2
>>> 
>>> req = urllib2.Request("http://t.co/sJerAJfnH0")
>>> req.add_header('User-Agent', 'Mozilla/5.0 (Windows; U; Windows NT 5.1; en-GB; rv:1.9.0.3) Gecko/2008092417 Firefox/3.0.3')
>>> res = urllib2.urlopen(req)
>>> res.read()
'<noscript><META http-equiv="refresh" content="0;URL=http://klou.tt/t16tmmk50y9r"></noscript><title>http://klou.tt/t16tmmk50y9r</title><script>window.opener = null; location.replace("http:\\/\\/klou.tt\\/t16tmmk50y9r")</script>'
>>> 
```

**Comment by gmkou** [comment]
Fixed in 627f97b , #11
